### PR TITLE
Add to the context directory params for the Server and the Admin creation

### PR DESCRIPTION
**Pull request by morhag90**
Issue Number: #3767 

## PR Details

Add to the context directory params for the Server and the Admin creation.

## PR Checklist
- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-code-gen-types/src/plugin-events.ts [hunk 1]
```diff
@@ -59,4 +50,24 @@ export type Events = {
...
+    before?: (
+      dsgContext: DsgContext,
+      eventParams: CreateEntityServiceParams["before"]
+    ) => CreateEntityServiceParams["before"];
+    after?: (
+      dsgContext: DsgContext,
+      modules: CreateEntityServiceParams["after"]
+    ) => Module[];
+  };
+  [EventNames.CreateEntityBaseService]?: {
```
Comment (yuval-hazaz) [inline packages/amplication-code-gen-types/src/plugin-events.ts]: rename to CreateEntityServiceBase

File: packages/amplication-code-gen-types/src/plugin-events-params.ts [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
 
-export interface CreateServiceModulesParams extends EventParams {
+export interface CreateEntityBaseServiceParams extends EventParams {
   before: {
     entityName: string;
-    entityType: string;
     entity: Entity;
+    templateMapping: { [key: string]: any };
+    passwordFields: EntityField[];
+    srcDirectory: string;
```
Comment (yuval-hazaz) [inline packages/amplication-code-gen-types/src/plugin-events-params.ts]: please use the data from the context

File: packages/amplication-code-gen-types/src/plugin-events-params.ts [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+    serviceId: namedTypes.Identifier;
+    serviceBaseId: namedTypes.Identifier;
+    delegateId: namedTypes.Identifier;
+    entityDTO: NamedClassDeclaration;
+    dtos: DTOs;
```
Comment (yuval-hazaz) [inline packages/amplication-code-gen-types/src/plugin-events-params.ts]: please use the data from the context

File: packages/amplication-code-gen-types/src/plugin-events-params.ts [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
+    file: namedTypes.File;
+  };
+}
+
+export interface CreateEntityServiceParams extends EventParams {
+  before: {
+    entityName: string;
+    templateMapping: { [key: string]: any };
+    passwordFields: EntityField[];
     srcDirectory: string;
```
Comment (yuval-hazaz) [inline packages/amplication-code-gen-types/src/plugin-events-params.ts]: please use the data from the context

File: packages/amplication-data-service-generator/src/tests/__snapshots__/create-data-service-without-graphql.spec.ts.snap [hunk 5]
```diff
@@ -14482,6 +14376,112 @@ export class StringNullableFilter {
   @Type(() => String)
   not?: string;
 }
+",
+  "undefined/App.tsx": "import React, { useEffect, useState } from \\"react\\";
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/tests/__snapshots__/create-data-service-without-graphql.spec.ts.snap:14380]: something is wrong in the snapshot data - please note the URL starts with `undefined`
```suggestion
  "admin-ui/App.tsx": "import React, { useEffect, useState } from \\"react\\";
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-code-gen-types/src/plugin-events.ts (hunk 1 above)
Comment (morhag90) [reply packages/amplication-code-gen-types/src/plugin-events.ts]: @yuval-hazaz  also update the params object : from  CreateEntityBaseServiceParams to: CreateEntityServiceBaseParams

**Pull request merged**